sycl/doc/extensions/proposed/sycl_ext_intel_fpga_mem.asciidoc

```diff
@@ -0,0 +1,488 @@
+  inline constexpr port_direction_key::value_t<d> port_direction;
+  inline constexpr port_direction_key::value_t<port_direction_enum::uni>
+      port_direction_uni;
+  inline constexpr port_direction_key::value_t<port_direction_enum::bi>
+      port_direction_bi;
+
+  template <bool b>
+  inline constexpr clock_2x_key::value_t<b> clock_2x;
+  inline constexpr clock_2x_key::value_t<true> clock_2x_true;
+  inline constexpr clock_2x_key::value_t<false> clock_2x_false;
+
+  template <ram_stitching_enum d>
+  inline constexpr ram_stitching_key::value_t<d> ram_stitching;
+  inline constexpr ram_stitching_key::value_t<ram_stitching_enum::min_ram>
+      ram_stitching_min_ram;
+  inline constexpr ram_stitching_key::value_t<ram_stitching_enum::max_fmax>
+      ram_stitching_max_fmax;
+
+  template <size_t n>
+  inline constexpr private_copies_key::value_t<n> private_copies;
+
+  template <size_t n>
+  inline constexpr num_replicates_key::value_t<n> num_replicates;
+
+} // namespace sycl::ext::intel::experimental
+----
+
+[frame="topbot",options="header"]
+|===
+|Property |Description
+
+a|
+[source,c++]
+----
+resource
+----
+a|
+Specifies which FPGA memory resources to use to implement the variable.
+
+The following values are supported:
+
+* `mlab`: data is stored in special Adaptive Logic Modules (ALMs), called memory-logic array blocks 
+* `block_ram`: data is stored in dedicated block RAM modules, called M20K modules
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+num_banks
+----
+a|
+The number of banks that the array is divided into. If the `stride_size` property is not specified in conjunction with the `num_banks` property, the memory will be divided using a cyclic partitioning strategy, where consecutive words will be placed in different banks.
+
+Must be greater than zero and less than number of array elements.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+stride_size
+----
+a|
+The number of consecutive elements in an array that will be part of the same bank. If the `num_banks` property is not specified in conjunction with `stride_size` property, memory will be divided using block partitioning strategy, where consecutive words will be placed in the same bank until `stride_size` worth of elements is reached, at which point further words will be added to a new bank.
+
+Must be greater than zero and less than the number of array elements.
+
+If the `word_size` property is specified, `stride_size` must be a multiple of `word_size`. 
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+word_size
+----
+a|
+Size in elements of a single memory transaction.
+
+If the `stride_size` property is specified, `stride_size` must be a multiple of `word_size`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+port_direction
+----
+a|
+The following values are supported:
+
+* `uni`: Specifies that a port to the memory can only service read or write requests, but not both.
+* `bi`: Specifies that a port to the memory can service both read and write requests.
```

Comment:
Sorry if I'm contradicting my own previous opinion on this, but seeing a binary option like this use an enum seems silly when the language already supports "bool".  A property named "bi_directional_ports" with possible values of true or false seems simpler.